src/Umbraco.Core/IO/FileSystems.cs

```diff
@@ -38,92 +40,137 @@ public class FileSystems : IFileSystems
         #region Constructor
 
         // DI wants a public ctor
-        public FileSystems(IServiceProvider container, ILogger<FileSystems> logger, ILoggerFactory loggerFactory, IIOHelper ioHelper, IOptions<GlobalSettings> globalSettings, IHostingEnvironment hostingEnvironment)
+        public FileSystems(
+            ILoggerFactory loggerFactory,
+            IIOHelper ioHelper,
+            IOptions<GlobalSettings> globalSettings,
+            IHostingEnvironment hostingEnvironment,
+            IFileSystem stylesheetFileSystem)
         {
-            _container = container;
-            _logger = logger;
+            _logger = loggerFactory.CreateLogger<FileSystems>();
             _loggerFactory = loggerFactory;
             _ioHelper = ioHelper;
             _globalSettings = globalSettings.Value;
             _hostingEnvironment = hostingEnvironment;
+
+            _stylesheetsFileSystem = (ShadowWrapper)CreateShadowWrapper(stylesheetFileSystem, "css");
         }
 
-        // for tests only, totally unsafe
-        internal void Reset()
+        // Ctor for tests, allows you to set the various filesystems
+        internal FileSystems(
+            ILoggerFactory loggerFactory,
+            IIOHelper ioHelper,
+            IOptions<GlobalSettings> globalSettings,
+            IHostingEnvironment hostingEnvironment,
+            IFileSystem macroPartialFileSystem,
+            IFileSystem partialViewsFileSystem,
+            IFileSystem stylesheetFileSystem,
+            IFileSystem scriptsFileSystem,
+            IFileSystem mvcViewFileSystem) : this(loggerFactory, ioHelper, globalSettings, hostingEnvironment, stylesheetFileSystem)
         {
-            _shadowWrappers.Clear();
-            _filesystems.Clear();
-            Volatile.Write(ref _wkfsInitialized, false);
-            _shadowCurrentId = null;
+            _macroPartialFileSystem = (ShadowWrapper)CreateShadowWrapper(macroPartialFileSystem, "macro-partials");
+            _partialViewsFileSystem = (ShadowWrapper)CreateShadowWrapper(partialViewsFileSystem, "partials");
+            _scriptsFileSystem = (ShadowWrapper)CreateShadowWrapper(scriptsFileSystem, "scripts");
+            _mvcViewsFileSystem = (ShadowWrapper)CreateShadowWrapper(mvcViewFileSystem, "view");
+            // Set initialized to true so the filesystems doesn't get overwritten.
+            _wkfsInitialized = true;
+
         }
 
         // for tests only, totally unsafe
-        internal static void ResetShadowId()
-        {
-            _shadowCurrentId = null;
-        }
+        internal static void ResetShadowId() => _shadowCurrentId = null;
```

Comment:
Do we still need this? Ideally all static things are gone, if it's needed I don't think it needs to be static?

Reply:
I thought it was still required for a test, but I was wrong, I've removed it.